sentry-good_job/lib/sentry/good_job/cron_monitoring.rb

```diff
@@ -0,0 +1,158 @@
+          nil
+        end
+
+        # Generate monitor slug from job name
+        def self.monitor_slug(job_name)
+          job_name.to_s.underscore.gsub(/_job$/, "")
+        end
+
+        # Parse cron expression and extract timezone
+        def self.parse_cron_with_timezone(cron_expression)
+          return [cron_expression, nil] unless cron_expression && !cron_expression.strip.empty?
+
+          parts = cron_expression.strip.split(" ")
+          return [cron_expression, nil] unless parts.length > 5
+
+          # Last part might be timezone
+          timezone = parts.last
+          # Comprehensive timezone validation that handles:
+          # - Standard timezone names (UTC, GMT)
+          # - IANA timezone identifiers (America/New_York, Europe/Stockholm)
+          # - Multi-level IANA timezones (America/Argentina/Buenos_Aires)
+          # - UTC offsets (UTC+2, UTC-5, GMT+1, GMT-8)
+          # - Numeric timezones (GMT-5, UTC+2)
+          if timezone.match?(/^[A-Za-z_]+$/) || # Simple timezone names (UTC, GMT, EST, etc.)
+             timezone.match?(/^[A-Za-z_]+\/[A-Za-z_]+$/) || # Single slash timezones (Europe/Stockholm)
+             timezone.match?(/^[A-Za-z_]+\/[A-Za-z_]+\/[A-Za-z_]+$/) || # Multi-slash timezones (America/Argentina/Buenos_Aires)
+             timezone.match?(/^[A-Za-z_]+[+-]\d+$/) || # UTC/GMT offsets (UTC+2, GMT-5)
+             timezone.match?(/^[A-Za-z_]+\/[A-Za-z_]+[+-]\d+$/) # IANA with offset (Europe/Stockholm+1)
+            cron_without_timezone = cron_expression.gsub(/\s+#{Regexp.escape(timezone)}$/, "")
+            [cron_without_timezone, timezone]
+          else
+            [cron_expression, nil]
+          end
+        end
+      end
+
+      # Main integration class that handles all cron monitoring setup
+      # This class follows Good Job's integration patterns and Sentry's extension guidelines
+      class Integration
+        # Set up monitoring for all scheduled jobs from Good Job configuration
+        def self.setup_monitoring_for_scheduled_jobs
+          return unless ::Sentry.initialized?
+          return unless ::Sentry.configuration.good_job.auto_setup_cron_monitoring
+
+          cron_config = ::Rails.application.config.good_job.cron
+          return unless cron_config.present?
+
+          cron_config.each do |job_name, job_config|
+            setup_monitoring_for_job(job_name, job_config)
+          end
+
+          Sentry::GoodJob::Logger.info "Sentry cron monitoring setup for #{cron_config.keys.size} scheduled jobs"
+        end
+
+        # Set up monitoring for a specific job
+        def self.setup_monitoring_for_job(job_name, job_config)
+          job_class_name = job_config[:class]
+          cron_expression = job_config[:cron]
+
+          return unless job_class_name && cron_expression
+
+          # Get the job class
+          job_class = begin
+            job_class_name.constantize
```

Comment:
It would be nice if the job classes weren't constantized. If there's any chance this is initialized during development, this would negatively impact app boot.